devicetypes/smartthings/rgbw-light.src/rgbw-light.groovy

```diff
@@ -60,29 +100,54 @@ metadata {
 		}
 	}
 
-	standardTile("reset", "device.reset", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
-		state "default", label:"Reset Color", action:"reset", icon:"st.lights.philips.hue-single"
-	}
-
 	controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range:"(2700..6500)") {
 		state "colorTemperature", action:"color temperature.setColorTemperature"
 	}
 
 	main(["switch"])
-	details(["switch", "levelSliderControl", "rgbSelector", "colorTempSliderControl", "reset"])
+	details(["switch", "levelSliderControl", "colorTempSliderControl"])
 }
 
+private getCOLOR_TEMP_MIN() { 2700 }
+private getCOLOR_TEMP_MAX() { 6500 }
+// For Z-Wave devices, we control illumination by crossfading the cold and warm
+// white channels.  But for devices that only have single cold or warm white
+// illumination (as with many RGBW LED strips), we cannot dim either white
+// channel to 0, as this will then completely turn off illumination.
+// Therefore, we lower-bound both white channels to 1.  This will have no
+// perceptible impact for devices that actually support white temperature
+// cross-fading, but will keep devices that do not doing something sane from
+// the user perspective, which is to modify intensity for the single white
+private getWHITE_MIN() { 1 } // min for Z-Wave coldWhite and warmWhite paramaeters
+private getWHITE_MAX() { 255 } // max for Z-Wave coldWhite and warmWhite paramaeters
+private getCOLOR_TEMP_DIFF() { COLOR_TEMP_MAX - COLOR_TEMP_MIN }
+private getRED() { "red" }
+private getGREEN() { "green" }
+private getBLUE() { "blue" }
+private getWARM_WHITE() { "warmWhite" }
+private getCOLD_WHITE() { "coldWhite" }
+private getRGB_NAMES() { [RED, GREEN, BLUE] }
+private getWHITE_NAMES() { [WARM_WHITE, COLD_WHITE] }
+private getCOLOR_NAMES() { RGB_NAMES + WHITE_NAMES }
+private getSWITCH_VALUE_ON() { 0xFF } // Per Z-Wave, this multilevel switch value commands state-transition to on.  This will restore the most-recent non-zero value cached in the device.
+private getSWITCH_VALUE_OFF() { 0 } // Per Z-Wave, this multilevel switch value commands state-transition to off.  This will not clobber the most-recent non-zero value cached in the device.
+private MIN(a, b) { a < b ? a : b }
+private MAX(a, b) { a > b ? a : b }
+private BOUND(x, floor, ceiling) { x < floor ? floor : x > ceiling ? ceiling : x }
```

Comment:
Isn't this basically the same as `Math.max(Math.min(x, ceiling), floor)`? Would it make sense to use one or the other? (You have `Math.max(Math.min(level, 99), 1)` below, which I am thinking could just be `BOUND(level, 1, 99)` and `BOUND` either being what it is or the `Math.min`+`Math.max`.)

Reply:
Ah, fair enough.  Yes, this is redundant.  Will fixup.